src/ToolBox/superpmi/superpmi-shared/methodcontext.cpp

```diff
@@ -6061,3 +6061,46 @@ int MethodContext::dumpMD5HashToBuffer(BYTE* pBuffer, int bufLen, char* hash, in
 
 #endif // !FEATURE_PAL
 }
+
+DenseLightWeightMap<MethodContext::Agnostic_Environment>* MethodContext::prevEnviroment = nullptr;
+
+bool MethodContext::wasEnviromentChanged()
+{
+    bool changed = false;
+    if (prevEnviroment == nullptr)
+    {
+        changed = true;
+    }
+    else if (Environment->GetCount() != prevEnviroment->GetCount())
+    {
+        changed = true;
+    }
+    else
+    {
+        for (unsigned int i = 0; i < Environment->GetCount(); i++)
+        {
+            Agnostic_Environment currEnvValue = Environment->Get(i);
+            LPCSTR currKey = (LPCSTR)Environment->GetBuffer(currEnvValue.name_index);
+            LPCSTR currVal = (LPCSTR)Environment->GetBuffer(currEnvValue.val_index);
+
+            Agnostic_Environment prevEnvValue = prevEnviroment->Get(i);
```

Comment:
I'm not 100% familiar with the implementation of these maps, but is it a guarantee that if the same key/value pair is present in two map instances that it will have the same index in each?

Reply:
It is based on GetEnvironmentStrings, that returns a environment block that guarantee to be ordered.
> All strings in the environment block must be sorted alphabetically by name. The sort is case-insensitive, Unicode order, without regard to locale. (https://msdn.microsoft.com/en-us/library/windows/desktop/ms682009(v=vs.85).aspx)

Reply:
That isn't making a statement about the map itself, though. This is really a question about the implementation of `DenseLightWeightMap`, not about the source of the data that fills it.

Reply:
Because we insert them in sorted order they will be in sorted order.

Reply:
Okay, great.